arch/simddetect.cpp

```diff
@@ -26,7 +26,7 @@
 #endif // x86 target
 
 #if defined(X86_BUILD)
-# if defined(__linux__) || defined(__MINGW32__) || defined(__APPLE__)
+# if defined(__GNUC__) || defined(__clang__) || defined(__MINGW32__) || defined(__MINGW64__) 
```

Comment:
I'm aware of what GNUC means but it's not obvious and unfortunate that other compilers use it as well. I think  `defined(__GNUC__) || defined(__clang__)` is more explicit.

In the interest of moving forward I made these change with some comments to note the implications for Mingw64 and clang. 

Checked with gcc and clang.